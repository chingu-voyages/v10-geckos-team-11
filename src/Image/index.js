import React from 'react'

const Image = ({ float, noResponsive, noBorder, src }) => {
  const imageClass = []

  if (float === 'left' || float === 'right') {
    imageClass.push(`float-${float}`)
  }

  if (noResponsive) {
    imageClass.push('no-responsive')
  }

  if (noBorder) {
    imageClass.push('no-border')
  }

  const className = imageClass.length > 0 ? imageClass.join(' ') : undefined

  return <img className={className} src={src} />
}

export default Image
