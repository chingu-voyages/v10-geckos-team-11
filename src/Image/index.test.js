import React from 'react'
import renderer from 'react-test-renderer'

import Image from '.'

describe('Image', () => {
  it('renders default image with border', () => {
    const img = renderer.create(<Image />).toJSON()

    expect(img).toMatchSnapshot()
  })

  it('`noResponsive` prop renders img tag with `no-responsive` class', () => {
    const imgNotResponsive = renderer.create(<Image noResponsive />).toJSON()

    expect(imgNotResponsive.props.className).toBe('no-responsive')
  })

  it('`noBorder` prop renders img tag with `no-border` class', () => {
    const imgNoBorder = renderer.create(<Image noBorder />).toJSON()

    expect(imgNoBorder.props.className).toBe('no-border')
  })

  it('`float` prop set to "left" or "right" sets the float className', () => {
    const imgFloatedRight = renderer.create(<Image float="right" />).toJSON()
    const imgFloatedLeft = renderer.create(<Image float="left" />).toJSON()

    expect(imgFloatedRight.props.className).toBe('float-right')
    expect(imgFloatedLeft.props.className).toBe('float-left')
  })
})
