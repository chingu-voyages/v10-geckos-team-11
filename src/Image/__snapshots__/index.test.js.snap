// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders default image with border 1`] = `<img />`;
